### PR TITLE
Allow set of vehicle origin even when using GPS

### PR DESCRIPTION
**Pull request by peterbarker**
not being able to set origin when using GPS is annoying.  We've adjusted the way we do positions in the EKF (local offsets) and generally (Vector3p), so even if the suer sets an origin a long way away it shouldn't hurt.

Some interesting results here:

image

vs the same test run without the offset:
image

### CODE REVIEW

**Comment by rmackay9** [comment]
I agree it should be OK to set the origin now.  This check was added back before we had the moving internal origin.

File: libraries/AP_NavEKF3/AP_NavEKF3.cpp [hunk 1]
```diff
@@ -1392,12 +1392,10 @@ bool NavEKF3::setOriginLLH(const Location &loc)
     if (!core) {
         return false;
     }
-    if ((sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS) || common_origin_valid) {
-        // we don't allow setting of the EKF origin if using GPS
-        // or if the EKF origin has already been set.
-        // This is to prevent accidental setting of EKF origin with an
-        // invalid position or height or causing upsets from a shifting origin.
-        GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "EKF3 refusing set origin");
+    if (common_origin_valid) {
```
Comment (tridge) [inline libraries/AP_NavEKF3/AP_NavEKF3.cpp:1395]: I don't think this will work as NavEKF3_core::setOriginLLH() also checks

**Review by tridge — DISMISSED** [review]

File: libraries/AP_NavEKF3/AP_NavEKF3.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_NavEKF3/AP_NavEKF3.cpp:1395]: This is resolved now because PeterB has removed the lines from NavEKF3_core::setOriginLLH() as well now.

**Review by rmackay9 — CHANGES_REQUESTED** [review]
I very much like this change but I think we should remove the EKF2 changes to reduce testing effort.

**Comment by rmackay9** [comment]
Hi @peterbarker,

Somehow the change to AP_NavEKF3_Control.cpp has gone missing or perhaps I misread the the change in it's earlier form.  In any case, maybe you could pull that removal in as well from my PR?  https://github.com/ArduPilot/ardupilot/pull/27081

**Comment by peterbarker** [comment]
> I very much like this change but I think we should remove the EKF2 changes to reduce testing effort.

Done

**Pull request closed without merging**